HLTrigger/Configuration/python/Phase2/hltPhase2Setup_cff.py

```diff
@@ -0,0 +1,9996 @@
+import FWCore.ParameterSet.Config as cms
```

Comment:
Sorry, but I do not like a 10000 line dump without knowing how it was arrived at, and how it is going to be maintained from release to next release. For example, is there an area in ConfDB which contains this part?

Reply:
I completely agree that a 10000 line dump is really bad. So, the rationale was to have something close in spirit to the `hltGetConfiguration --setup`, but of course that reads from ConfDB, which we are not using (and do not plan to use for the time being). On the other hand, we want to have *all* of the code for the HLT Phase 2 TDR (and that should eventually evolve into the HLT Phase 2 code) in the release -- i.e. minimize the usage of private areas, and make sure it is easily reproducible in the future. To answer your questions:

1) This file was arrived at by taking the Baseline Tracking configuration, fully expanding it (to get rid of Tasks, Placeholders, Eras, customisations...), and splitting the ESSources, ESProducers off.

2) How is going to be maintained release through release: I reckon that at some point, near the end of the integration, this should be part of a workflow that runs something like an "HLT:Phase2" step, and we will monitor that for any regression.

Some ideas that I am having as I type:
a) We could split the big file into a lot of cfis (I checked and it expands to 734 cfis). But there is a lot of unneeded stuff (at least 300 of those are different jet configurations, for instance). Splitting and cleaning could be a first step here, what do you think?
b) Getting most of those ESSources and ESProducers from their offline setups seems a good idea. I would be worried, however, with offline cffs changing under our feet. Having *all* the needed configurations in our area in the release, at least until the TDR is released, seemed like a better option to me.

Cheers,
Thiago

Reply:
Okay, I split the big file into a small file that only loads some 100 `cfi` or so. The `cfi`s are going to be treated in one of the ways discussed below:

- the autogenerated cfi file
- a fully consistent cfi file maintained by hand (for Phase 2)
- a cfi file that defines the Phase 2 configuration using Eras (this will require setting the right Era in the HLT configuration)

But at least the issue of being an unmaintainable dump seems solved. I am marking this conversation as resolved.